SamplesV2/ContinuousIntegrationAndDelivery/PrePostDeploymentScript.Ver2.ps1

```diff
@@ -0,0 +1,675 @@
+}
+
+$templateParameters = $null
+if (Test-Path -Path $ArmTemplateParameters) {
+    $templateParametersJson = Get-Content $ArmTemplateParameters | ConvertFrom-Json
+    $templateParameters = $templateParametersJson.parameters
+}
+
+#Triggers 
+Write-Host "Getting triggers"
+$triggersInTemplate = $resources | Where-Object { $_.type -eq "Microsoft.DataFactory/factories/triggers" }
+$triggerNamesInTemplate = $triggersInTemplate | ForEach-Object { $_.name.Substring(37, $_.name.Length - 40) }
+
+$triggersDeployed = Get-SortedTriggers -DataFactoryName $DataFactoryName -ResourceGroupName $ResourceGroupName
+
+if ($PreDeployment -eq $true) {
+    #Stop trigger only if there is change in payload
+    $triggersToStop = $triggersDeployed | Where-Object { $_.Name -in $triggerNamesInTemplate -and $_.RuntimeState -ne 'Stopped' } `
+    | Where-Object { $triggerName = $_.Name;
+        $triggerInTemplate = $triggersInTemplate | Where-Object { $_.name.Substring(37, $_.name.Length - 40) -eq $triggerName };
+        Compare-TriggerPayload -triggerDeployed $_ -triggerInTemplate $triggerInTemplate -templateParameters $templateParameters
+    } `
+    | ForEach-Object { 
+        New-Object PSObject -Property @{
+            Name        = $_.Name
+            TriggerType = $_.Properties.GetType().Name 
+        }
+    }  
+
+    Write-Host "Stopping $($triggersToStop.Count) triggers  `n"
+    $triggersToStop | ForEach-Object {
+        if ($_.TriggerType -eq "BlobEventsTrigger") {
+            Write-Host "Unsubscribing" $_.Name "from events"
+            $status = Remove-AzDataFactoryV2TriggerSubscription -ResourceGroupName $ResourceGroupName -DataFactoryName $DataFactoryName -Name $_.Name
+            while ($status.Status -ne "Disabled") {
+                Start-Sleep -s 15
+                $status = Get-AzDataFactoryV2TriggerSubscriptionStatus -ResourceGroupName $ResourceGroupName -DataFactoryName $DataFactoryName -Name $_.Name
+            }
+        }
+        Write-Host "Stopping trigger" $_.Name
+        Stop-AzDataFactoryV2Trigger -ResourceGroupName $ResourceGroupName -DataFactoryName $DataFactoryName -Name $_.Name -Force
+    }
+
+    $explicitTriggersToStop = $triggersDeployed | Where-Object { $_.Name -in $triggerNamesInTemplate -and $_.RuntimeState -ne 'Stopped' } `
+    | Where-Object { $_.Name -in $ExplicitStopTriggerList } `
+    | ForEach-Object { 
+        New-Object PSObject -Property @{
+            Name        = $_.Name
+            TriggerType = $_.Properties.GetType().Name 
+        }
+    }
+    
+    if ($explicitTriggersToStop -and $explicitTriggersToStop.Count -gt 0) {
+        Write-Host "Stopping $($explicitTriggersToStop.Count) triggers from explicit stop-trigger list `n"
+        $explicitTriggersToStop | ForEach-Object {
+            if ($_.TriggerType -eq "BlobEventsTrigger") {
+                Write-Host "Unsubscribing" $_.Name "from events"
```

Comment:
533 line is fine but 536 is not :( 
`Write-Host "Unsubscribing '$($_.Name)' from events"`

Reply:
same at lines 543, 591, 595, 608, 613, 618, 623, 628, 633, 638, 643, 646, 665, 672
line 661 is fine